modal/_utils/blob_utils.py

```diff
@@ -497,7 +503,11 @@ def hash_block_sha256(block_idx: int) -> bytes:
 
         return sha256_hash.digest()
 
-    tasks = (asyncio.to_thread(functools.partial(hash_block_sha256, idx)) for idx in range(num_blocks))
+    async def hash_block_sha256(block_idx: int) -> bytes:
+        async with hash_semaphore:
+            return await asyncio.to_thread(functools.partial(blocking_hash_block_sha256, block_idx))
```

Comment:
Unrelated to this PR but do you know why we don't pass args directly to `to_thread`?

Reply:
Ah nice, didn't know that was a thing!